Mathlib/Analysis/Fourier/DirichletKernel.lean

```diff
@@ -0,0 +1,65 @@
+/-
+Copyright (c) 2025 Yi Yuan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yi Yuan
+-/
+module
+
+public import Mathlib.Analysis.Complex.Trigonometric
+
+/-!
+# Dirichlet kernel
+
+The Dirichlet kernel is a family of functions related to the Fourier Transform. This file defines lemmas relevant to the Dirichlet kernel.
+
+## References
+
+* https://en.wikipedia.org/wiki/Dirichlet_kernel
+
+
+## TODO
+
+- Define the Dirichlet kernel
+
+- Show that $S_n(f)(x) = (D_n * f)(x)$
+
+-/
+
+@[expose] public section
+
+open Real
+
+section
+
+theorem sum_range_cos_mul_add (n : ℕ) (x a : ℝ) (hx : sin (x / 2) ≠ 0) :
+    ∑ k ∈ Finset.range n, cos ((k : ℝ) * x + a)
+    = sin (n * x / 2) / sin (x / 2) * cos ((n - 1 : ℝ) * x / 2 + a) := by
+  induction n with
+  | zero => simp
+  | succ n ih =>
+    simp [Finset.sum_range_succ, ih, field]
```

Comment:
AFAIR, we prefer to avoid non-closing `simp`s, unless this changed recently. You can use, e.g., `simp only` or `simp? [...] says simp only [...]` instead.